### ISSUE TITLE
[ScriptableObjectFieldInspector.cs] List of issues for this script

### ISSUE THREAD

**Issue by b3x206** [b3x206/bxfw-unity#15]
## Alright. Where to begin?

> I made the `ScriptableObjectFieldInspector` as a convenience, but it ended up being a burden (well, sort of. it is a great convenience when it is, such as allowing easy editing of inline serialized scriptable objects). But now, i realized how broken the script was.

* Issue 1:
Now when you use this FieldInspector with unity ReorderableList arrays, everything goes to hell.
This 'ReorderableList' thing, does everything without my dumb script knowing anything.
So if it changed the size, there are two potential things happening :
&nbsp;&nbsp;&nbsp;&nbsp;1. If the array shrinked (size is lower, 1 SerializedProperty reference lost), if the indexes don't align like it is sequential, the script will just tell me, on an unrelated method trying to use the `SerializedProperty` ```no whowopsies nowo i made a ewxcepwtion wowo nya pwopewty twhat was at remwovoed iwndex disawpeaw??``` [this thread helped, because the error is very undescriptive](https://forum.unity.com/threads/property-drawer-serializedproperty-disappeared.528373/) and it just broke everything
&nbsp;&nbsp;&nbsp;&nbsp; or
&nbsp;&nbsp;&nbsp;&nbsp;2. Array has grew in size and we added an element. hooray everyone is happy, because i get to draw my null inspector and unity gets to allocate me a SerializedProperty for the implemented `PropertyDrawer`.
Now we can't control whether the given `SerializedProperty` is valid, so this needs a proper solution and debugging.

* Issue 2:
This in the ```protected void HandleDifferentDrawers()``` method and when you move an element in the `ReorderableList` it just keeps cloning the moved item and yeah, that is the issue. Most likely an easily (or hardly) fixable issue.

* Issue idk (not really an issue):
GUI code sucks (but the GUI works fine so won't touch it)

Basically this script has given me pain. But it will be worth when i fix array related issues.
damn arrays, always changing without telling

**Comment by b3x206** [comment]
huh, fixing issue 2 did fix 1 as well.
now i just have to deal with the unity's crappy serializer and it's respective 'SetDirty' method being not very functional

**Comment by b3x206** [comment]
**A new 3rd issue has appeared**

Fortunately, even though it was painful to debug for the solution was relatively easy
When we add the creation delegate event to the `GenericMenu`, i passed the `SerializedProperty` that we get from `GetPropertyHeight`. Apparently that `SerializedProperty` is disposed automatically [by design](https://issuetracker.unity3d.com/issues/argumentnullexception-exception-is-thrown-on-unity-self-when-using-serializedobject-dot-targetobject-inside-a-delegate) (i mean, it makes sense), but i just googled the cryptic error message that appears and thankfully there was an explanation. When the creation delegate was called, since the property was disposed we had lots of error.

* First i tried `property.Copy()` but that didn't clone the parent `SerializedObject`, which apparently we need.
* Then i manually created a `SerializedObject` then called the `FindProperty` with the given serialized property path of our `property` parameter and it worked.

So yeah this the 3rd issue that probably exists, next commit should fix it

**Comment by b3x206** [comment]
Closing #15 as it's no longer needed
The class, while has a custom inspector support badly implemented (fix is still todo), we don't have serious issues for now

Added a custom gui command dictionary as a last closing note for this script for a while.
